Reporting/M5 Document/Reflection.md

```diff
@@ -0,0 +1,29 @@
+## How did your project management work for the team?  What was the hardest thing, and what would you do the same/differently the next time you plan to complete a project like this? 
+
+- Our project management worked relatively well for the team. We maintained regular communication channels, utilized project management tools effectively, and delegated tasks efficiently. However, the hardest aspect was ensuring effective coordination between front-end and back-end development teams, especially when integrating complex features like peer-to-peer messaging and encryption. 
+
+- In future projects, we would prioritize more frequent cross-team check-ins and perhaps consider adopting a more integrated development approach from the outset to mitigate such challenges.
+
+## Do you feel that your initial requirements were sufficiently detailed for this project?  Which requirements did you miss or overlook?
+- While our initial requirements provided a solid foundation for the project, there were some areas where additional detail could have been beneficial. Specifically, we overlooked the specifics of password security measures, and multimedia messaging capabilities. 
```

Comment:
Correction:
- Password security is enterprise-grade in our application because we are using an authentication provider. We are using webhooks to sync our MongoDB data with our provider's database, and we are not storing any passwords in our database. The provider takes care of everything security-related in the area of authentication
- The core problem with our initial requirements revolved around the distinction between admins, owners and members. This was unclear due to us misunderstanding how our server channel system should work, as we only found out towards the final week that we don't need to implement the idea of separate server channels. This caused a significant number of uncertainties in the types of privileges that admins would get